lib/utils/checker.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"net"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/gravitational/trace"
+)
+
+type CertChecker struct {
```

Comment:
I'm afraid it can break some cases like Teleport recording proxy mode with custom keys loaded to the agent. Instead, why can't we make a list of allowed algos and ciphers a parameter to the cert checker and in FIPS mode explicitly instantiate the checker in the services module like we do with everything else?

Reply:
@klizhentas It will not break custom keys. With agent forwarding, requests will be forwarded to the agent running on your client and OpenSSH `ssh-agent`, simply tries all keys loaded into the agent.

For users using OpenSSH `ssh` and `ssh-agent`, nothing should change.